src/components/WeatherComponent.tsx

```diff
@@ -1,34 +1,19 @@
-import axios from 'axios';
-import { Weather } from '../types';
-import UserForm from './UserForm';
+import { WeatherContext } from './context/WeatherContext';
 import { Title, Text, Container } from '@mantine/core';
-import { useEffect, useState, createContext } from 'react';
-
-export const WeatherContext = createContext<any>(null);
+import { useContext } from 'react';
 
 const WeatherComponent = () => {
-  const [weather, setWeather] = useState<Weather | null>();
-  const fetchWeatherData = async () => {
-    const response = await axios.get('http://mock-api-call/weather/get-weather');
-
-    setWeather(response.data.result.weather);
-  };
-
-  useEffect(() => {
-    fetchWeatherData();
-  }, []);
+  const weather = useContext(WeatherContext);
```

Comment:
to make things a little cleaner you can deconstruct the object coming from the context like this
```javascript
const { weather } = useContext(WeatherContext);
```
that way you don't need to do `weather.weather`